### ISSUE TITLE
Suggestion：Space Ship and Interstellar Travel

### ISSUE THREAD

**Issue by FuYang001** [Advanced-Rocketry/AdvancedRocketry#1229]
When I played this mod, the best thing I found is that I can add my custom planets and there's a variety of concepts I can use. For example, in Advanced Planet Configuration, I can set many content like OreGen and Biomes to create my own planet. Even more details like artifact, dim mapping I can modify, too. No double they are absolutely very useful for those who want to make their own unique modpack.

But, the question is, rockets can be built out of any block in the game, which make me so confused. In my opinion, rocket should be built using various rocket parts like frame, nose cone, I should have become ash when launching the rocket without any shelter and necessary protection, with rocket crash.

I hope there could be function that rocket's space capabilities (like range limitaion) are depended on what blocks are used in building, not just have the same space capabilities in any rockets. I also hope that it could be configurable because it must be very useful to pack makers.

In the real world, interstellar travel is far more to us now, I think chemical rocket can't supports us to do that, I hope there could be some new aerospace vehicles like space ship, which is more resonable to perform interstellar travel. Or other vehicles. Whatever it is, it must be some later game stage things, so it can be some kind of hard to craft.

Another question is, as we all knew, the interstellar travel in the game is just dimension teleport, GC is also the same. So I thought, could there be a astronavigation in the game? It is depended on the distance between two planets, asteriod's density in the universe, etc. Maybe it could be using a special dimension or other more efficient way to achieve.

More importantly, now, the space station in game can be built in once, like squeeze a building in a little box. In my view, space station should be built step by step, not just in once and the same as rocket, which can be built out of any block in the game. A space station should also needs a variety of  necessary components and modules to support its operation, such as solar panel, Core Module, etc.

Actually, I'm a pack maker, so I hope the future contents could be configurable as many as possible. Maybe these things are hard to achieve, but if something I can help, let me do it.

**Comment by C1ffisme** [comment]
> But, the question is, rockets can be built out of any block in the game, which make me so confused. In my opinion, rocket should be built using various rocket parts like frame, nose cone, I should have become ash when launching the rocket without any shelter and necessary protection, with rocket crash.

I would agree with you, I actually like the idea of assembling parts Kerbal Space Program-style, but it's a bit too late to make that change now... (Which is a shame, because if you're limited to only a few certain parts, that means we can give each part mass, aerodynamic interaction, etc. It's a bit harder to do that with blocks.)

> In the real world, interstellar travel is far more to us now, I think chemical rocket can't supports us to do that, I hope there could be some new aerospace vehicles like space ship, which is more resonable to perform interstellar travel. Or other vehicles. Whatever it is, it must be some later game stage things, so it can be some kind of hard to craft.

The Space Station is capable of interstellar travel. If you tried to fly to another star in a tiny rocket cockpit, you'd be cramped for a very long time. Although, it would be cool to eventually allow smaller, vertical space fighters to go through hyperspace, but we don't have manual vertical spacecraft yet.

> Another question is, as we all knew, the interstellar travel in the game is just dimension teleport, GC is also the same. So I thought, could there be a astronavigation in the game? It is depended on the distance between two planets, asteriod's density in the universe, etc. Maybe it could be using a special dimension or other more efficient way to achieve.

For interplanetary travel, this would be pretty cool. For interstellar travel, though, I'd prefer it to stay where it is. It would be pretty cool if you could travel to other planets using Newtonian physics though...

> More importantly, now, the space station in game can be built in once, like squeeze a building in a little box. In my view, space station should be built step by step, not just in once and the same as rocket, which can be built out of any block in the game. A space station should also needs a variety of necessary components and modules to support its operation, such as solar panel, Core Module, etc.

[This is already possible.](http://65.78.115.159:8000/AdvancedRocketry/blocks/StationDockingPort.php)

**Comment by FuYang001** [comment]
>I would agree with you, I actually like the idea of assembling parts Kerbal Space Program-style, but it's a bit too late to make that change now... (Which is a shame, because if you're limited to only a few certain parts, that means we can give each part mass, aerodynamic interaction, etc. It's a bit harder to do that with blocks.)

I have a idea that using modularization building. First, we need to build all the necessary parts, like multiblocks, then "build" them into models like current rocket. Second, we need to assembly them together step by step, eventually become a bigger rocket model, which can start its performance. The modules can be various, and the modules' performance can be different in same types. If we don't assembly them in normal structure, it will be unstable even crash.

>The Space Station is capable of interstellar travel. If you tried to fly to another star in a tiny rocket cockpit, you'd be cramped for a very long time. Although, it would be cool to eventually allow smaller, vertical space fighters to go through hyperspace, but we don't have manual vertical spacecraft yet.

Oh, acutually, I mean. I think rocket should be capable of interstellar travel at least (like what I said, its performance should be depended on its modules), though warp is the only way to do that. But warp, a kind of Sci-Fi-like thing, should be far more later than rocket in game stage. 
About vertical spacecraft, I hope there could be some kind thing like that in the future, using modularization building, too. Its performance also depended on its modules.

>For interplanetary travel, this would be pretty cool. For interstellar travel, though, I'd prefer it to stay where it is. It would be pretty cool if you could travel to other planets using Newtonian physics though...

Yes, in all space mods, most of them don't have a astronavigation process. In my view, process is also very important in game play, it's not just waste time, there could be more chances to do things in space.

>This is already possible.

In fact, I mean, I want space station be built in modularization, through rocket's transportation to perform. And rocket's transportation capabilities should be depended on its modules, too. If a space station miss the necessary modules it can't perform normally, such as it can't warp. Now, space station's limit size is depended on the dock, which means once we put enough docks, we can build a ridiculously big space station like 32*32. Players just need to build space station in once, then get to it to build the rest of things themselves. Docking port is hardly been used.

In short, for rocket and other space craft, also space station, their performance should be depended on their structure stability and single module's quality.

**Comment by FuYang001** [comment]
@zmaster587

**Comment by voidsong-dragonfly** [comment]
Closed due to being moved to a central issue (#2036)

**Comment by JovietUnion** [comment]
Good. After all, this DOES look like a genuine effort at a Minecraft-style KSP RO recreation.
